crates/ruff_linter/src/rules/flake8_pyi/rules/custom_type_var_return_type.rs

```diff
@@ -79,46 +81,51 @@ impl Violation for CustomTypeVarReturnType {
 
 /// PYI019
 pub(crate) fn custom_type_var_return_type(
-    checker: &mut Checker,
-    function_def: &ast::StmtFunctionDef,
-) {
+    checker: &Checker,
+    binding: &Binding,
+) -> Option<Diagnostic> {
+    if !matches!(binding.kind, BindingKind::FunctionDefinition(_)) {
+        return None;
+    }
+
+    let semantic = checker.semantic();
+
+    let current_scope = &semantic.scopes[binding.scope];
+
+    let function_def = binding.statement(semantic)?.as_function_def_stmt()?;
```

Comment:
nit: not really sure we need all these blank lines; the code here isn't that complex

```suggestion
    let semantic = checker.semantic();
    let current_scope = &semantic.scopes[binding.scope];
    let function_def = binding.statement(semantic)?.as_function_def_stmt()?;
```